lib/can-npm-publish.js

```diff
@@ -37,9 +37,9 @@ const checkPrivateField = packagePath => {
  * @param packageName
  * @returns {Promise}
  */
-const viewPackage = packageName => {
+const viewPackage = (packageName, registry) => {
     return new Promise((resolve, reject) => {
-        const view = spawn("npm", ["view", packageName, "versions", "--json"]);
+        const view = spawn("npm", ["view", packageName, "versions", "--json", registry && `--registry=${registry}`]);
```

Comment:
Can you change this as follows?

```js
const registryArgs = registry ? [`--registry=${registry}`] : [];
const view = spawn("npm", ["view", packageName, "versions", "--json"].concat(registryArgs));
```

Because, I not want to put `undefined` value into `spawn`'s arguments.
```
var registry = undefined;
["a", registry && `--registry=${registry}`];
// ["a", undefined]
var registryArgs = registry ? [`--registry=${registry}`] : [];
["a"].concat(registryArgs);
// ["a"]
```

Reply:
Yes, thats definatly better.